src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,237 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private int capacity;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
 
     @Override
     public void put(K key, V value) {
-
+        if (table == null || size > threshold) {
+            resizeTable();
+        }
+        if (key == null) {
+            keyNullCase(value);
+        } else if (nodeBusyAndKeysEquals(key)) {
+            table[bucketIndex(key)].value = value;
+        } else if (nodeBusyAndKeysDifferent(key)) {
+            if (noCollision(key)) {
+                table[bucketIndex(key)].next = addNode(key,value);
+                size++;
+            } else {
+                putWithCollision(key,value);
+            }
+        } else {
+            table[bucketIndex(key)] = addNode(key,value);
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (size > 0) {
+            Node<K,V> node = table[bucketIndex(key)];
+            if (node != null) {
+                if (keysEquals(node,key)) {
+                    return node.value;
+                } else {
+                    return searchForCollision(node, key);
+                }
+            }
+        }
         return null;
     }
 
+    private void resizeTable() {
+        Node<K,V>[] newTable;
+        if (size > threshold) {
+            capacity = capacity * 2;
+            threshold = (int) (capacity * DEFAULT_LOAD_FACTOR);
+            newTable = (Node<K,V>[])new MyHashMap.Node[capacity];
```

Comment:
see example how to do it easy and without many private methods
![image](https://github.com/mate-academy/jv-hash-map/assets/20326858/d7f0c70f-85bc-4280-90cf-0009e727b3cf)